agent/src/agent/pipeline/pipeline.py

```diff
@@ -448,3 +446,27 @@ class PipelineRetries(Entity):
     def __init__(self, pipeline_: Pipeline):
         self.pipeline_id = pipeline_.name
         self.number_of_error_statuses = 0
+
+
+class PipelineMetric(object):
+    REQUIRED_KEYS = ['counters']
+
+    def __init__(self, metrics_: dict):
+        for key in self.REQUIRED_KEYS:
+            if key not in metrics_:
+                raise KeyError(f'The key `{key}` not in input dict')
+        self.stats = {
```

Comment:
there's no point in keeping these values in a dict, you can use regular attributes